v_2_0_0/master_template.go

```diff
@@ -1735,6 +1735,22 @@ write_files:
     [Service]
     ExecStartPre=/bin/bash -c "while [ ! -f /etc/audit/rules.d/10-docker.rules ]; do echo 'Waiting for /etc/audit/rules.d/10-docker.rules to be written' && sleep 1; done"
 
+- path: /etc/kubernetes/encryption/k8s-encryption-config.yaml
+  owner: root
+  permissions: 600
+  content: |
+    kind: EncryptionConfig
+    apiVersion: v1
+    resources:
+      - resources:
+        - secrets
+        providers:
+        - aescbc:
+            keys:
+            - name: key1
+              secret: {{.Extension.Keys.APIServerEncryptionKey | printf "%s" }}
+        - identity: {}
+
```

Comment:
Why does this change here? v2 is already in use in AWS AFAIK. So it means it must not change anymore. IMO v3 is the place to get changes in.

Reply:
ok lets move it to v3 so

Reply:
done

Reply:
I see this first comment here. And this comment https://github.com/giantswarm/kvm-operator/pull/320#discussion_r157724047

It sounds to me like dev versions and snapshots of them is a good idea. Otherwise we'll have hard times which versions we "can" change.

Reply:
For me the "dev" version is always the latest one. Once it is frozen a new one has to be established so people do not have to think. This is why it does not matter how we name the folder.

Reply:
Let's say we have version `v_2_0_0` and `v_3_0_0` and we want to add a feature to `v_2_0_0`. This means we need to create a version `v_2_1_0` but we also have `v_3_0_0`. How to recognize that we can work on `v_2_1_0`?

Reply:
Why should we work in both? Doesn't sound right to me. There are multiple versions and one is "dev". That is it. You describe a scenario where there are two dev versions. This does not work IMO. When there is 2 and 3 it means two is frozen and 3 is dev. We never look back. We always only move forward.

Reply:
What if a customer is using `v_2_0_0` which runs kubernetes 1.8. And our latest (i.e. dev) is `v_3_0_0` which runs kubernetes 1.9. We noticed serious bug in kubernetes 1.8 and we have to update to 1.8.1. But for valid reasons customer can't switch to kubernetes 1.9 ATM. What should we do?

Reply:
Then we introduce a patch and freeze it immediately. 2.1 is then the patch and is not touched anymore after creation. 3 still remains the dev playground.